### PR TITLE
Make the sort behavior is same on 32bit and 64bit.

### PR DESCRIPTION
**Pull request by halx99**

### CODE REVIEW

File: cocos/2d/CCNode.h [hunk 1]
```diff
@@ -950,8 +955,8 @@ class CC_DLL Node : public Ref
             return (n1->_localZOrderAndArrival < n2->_localZOrderAndArrival);
         });
 #else
-        std::stable_sort(std::begin(nodes), std::end(nodes), [](_T* n1, _T* n2) {
-            return n1->_localZOrder < n2->_localZOrder;
+        std::sort(std::begin(nodes), std::end(nodes), [](_T* n1, _T* n2) {
+            return n1->_localZOrder < n2->_localZOrder || (n1->_localZOrder == n2->_localZOrder && n1->_getOrderOfArrival() < n2->_getOrderOfArrival() );
```
Comment (minggo) [inline cocos/2d/CCNode.h]: The codes looks good, but i am not sure the performance. May be we should keep `_orderArrival` on 32bit.

Comment (halx99) [reply cocos/2d/CCNode.h]: Yes, direct access the variable more better.

Comment (minggo) [reply cocos/2d/CCNode.h]: Yep, the bug is imported to improve performance for 64bit devices, but we can not slow down 32bit devices.

Comment (halx99) [reply cocos/2d/CCNode.h]: Let's change something.

**Comment by halx99** [comment]
@minggo please review the changes.

**Comment by minggo** [comment]
Thanks, but please send to v3 branch since Node affect all codes, i think it is better to have more time to test it. We will release v3.16 soon.

**Comment by halx99** [comment]
OK, do you known how to change this PR to v3 branch directly? I find it.

**Comment by minggo** [comment]
Sorry i don't know what you mean `directly`, but you can go to https://github.com/halx99/cocos2d-x/tree/patch-1 and click `Pull request`.

**Comment by halx99** [comment]
I change base of this PR to v3branch

**Comment by halx99** [comment]
You also can see there is a common storage for 32bit & 64bit, ONLY sort code is different:
https://github.com/halx99/cocos2d-x/pull/35

**Comment by minggo** [comment]
@halx99 i think this PR includes my other codes that are not related with this issue. I think you should checkout a branch based on v3 to do it. You can use `git cherry-pick` to pick commits your need.

**Comment by halx99** [comment]
ok

**Comment by minggo** [comment]
@halx99 this PR contains much unneeded codes, i will close it. Please send a new one, thanks.

**Pull request closed without merging**